File: 258_Add_Digits.c
int addDigits(int num)
{

    if(num/10 == 0) return num;

    int val = num;
    int digits_val = 0;

    while(val > 0)
    {
        digits_val += val%10;
        val /= 10;
    } 

    return digits_val/10 == 0? digits_val : addDigits(digits_val) ;

}

File: 258_Add_Digits.py
class Solution:
    def addDigits(self, num: int) -> int:

        val = num
        addval = 0
        if(len(str(num)) == 1): 
            return num

        while val > 0:           
            addval = addval + val%10
            val = int(val/10)


        if len(str(addval)) >= 2:
            return self.addDigits(addval)
        else:
            return addval